src/udx.c

```diff
@@ -1510,15 +1505,10 @@ process_sacks (udx_stream_t *stream, char *buf, size_t buf_len) {
 
 static void
 process_data_packet (udx_stream_t *stream, int type, uint32_t seq, char *data, ssize_t data_len) {
-  if (seq == stream->ack && type == UDX_HEADER_DATA) {
+  if (seq == stream->ack && type & UDX_HEADER_DATA) {
     // Fast path - next in line, no need to memcpy it, stack allocate the struct and call on_read...
     stream->ack++;
 
-    stream->write_wanted |= UDX_STREAM_WRITE_WANT_STATE;
-    if (stream->socket != NULL) {
-      update_poll(stream->socket);
-    }
-
```

Comment:
I think this is redundant - `write_wanted |= WRITE_WANT_STATE` is set later in process_packet if the packet type was DATA or END